esphome/components/thermostat/thermostat_climate.cpp

```diff
@@ -615,9 +621,11 @@ void ThermostatClimate::switch_to_mode_(climate::ClimateMode mode) {
   this->mode = mode;
   this->prev_mode_ = mode;
   this->prev_mode_trigger_ = trig;
+  if (publish_state)
+    this->publish_state();
 }
 
-void ThermostatClimate::switch_to_swing_mode_(climate::ClimateSwingMode swing_mode) {
+void ThermostatClimate::switch_to_swing_mode_(climate::ClimateSwingMode swing_mode, const bool publish_state) {
```

Comment:
```suggestion
void ThermostatClimate::switch_to_swing_mode_(climate::ClimateSwingMode swing_mode, bool publish_state) {
```

it's declared as `bool` in the header, so it should match here

Reply:
clang-tidy doesn't like when we put `const` in the header, though. See the failed test.

Reply:
I could make it not-const, but...it should be `const`...it shouldn't be changed in the method...but I'm not an expert in modern semantics. Is `const` bad these days?

Reply:
I just removed it...this is a pretty trivial use case, no strong feelings either way. clang-tidy doesn't like `const` in the header, which I get. I know in the function itself it's a copy of the data so `const` is pretty irrelevant in that regard, but it's personal preference to use `const` in cases like this to prevent silly mistakes like `=` vs. `==`. It can also help express intent. I'm cool either way and if you don't like it, I'm most definitely not here to argue. 😄

Reply:
My concern was that this might be undefined/implementation-defined behavior :) but perhaps cpp has rules that const/non-const params resolve to the same symbol (don't know enough about that)